wrapanapi/systems/virtualcenter.py

```diff
@@ -955,6 +955,17 @@ def _search_folders_for_vm(self, name):
 
         return vm
 
+    def create_folder(self, folder_name):
+        """Create a Vm folder under Datacenter"""
+        dc = self.get_obj(vim.Datacenter, "Datacenter")
+        folder = dc.vmFolder.CreateFolder(folder_name)
+        return folder
+
+    def destroy_folder(self, folder_name):
+        """Delete a Vm folder under Datacenter"""
+        folder = self._search_folders_for_vm(folder_name)
+        folder.Destroy()
```

Comment:
can it return result of destroy command ?

Reply:
@izapolsk 
- destroy command returns ```vim.Task``` and if tried deleting again the folder which does not exists then it gives exception.
```
In [4]: w.create_folder("ghubale_new")                                                                                                                                       
Out[4]: 'vim.Folder:group-v74958'

In [5]: w.destroy_folder("ghubale_new")                                                                                                                                      
Out[5]: 'vim.Task:task-243984'

In [6]: w.destroy_folder("ghubale_new")                                                                                                                                      
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-3e0637eee608> in <module>
----> 1 w.destroy_folder("ghubale_new")

~/cfme_workspace/integration_tests/wrapanapi/wrapanapi/systems/virtualcenter.py in destroy_folder(self, folder_name)
    975         """
    976         folder = self._search_folders_for_vm(folder_name)
--> 977         return folder.Destroy()
    978 
    979     def _build_filter_spec(self, begin_entity, property_spec):

AttributeError: 'NoneType' object has no attribute 'Destroy'
```

- If we catch destroy object then we can check its info:
```
In [11]: w.create_folder("ghubale_data")                                                                                                                                     
Out[11]: 'vim.Folder:group-v74960'

In [12]: des_obj = w.destroy_folder("ghubale_data")                                                                                                                          

In [13]: des_obj.info                                                                                                                                                        
Out[13]: 
(vim.TaskInfo) {
   dynamicType = <unset>,
   dynamicProperty = (vmodl.DynamicProperty) [],
   key = 'task-243988',
   task = 'vim.Task:task-243988',
   description = <unset>,
   name = vim.ManagedEntity.Destroy,
   descriptionId = 'Folder.destroy',
   entity = 'vim.Folder:group-v74960',
   entityName = 'ghubale_data',
   locked = (vim.ManagedEntity) [],
   state = 'success',
   cancelled = false,
   cancelable = false,
   error = <unset>,
   result = <unset>,
   progress = <unset>,
   reason = (vim.TaskReasonUser) {
      dynamicType = <unset>,
      dynamicProperty = (vmodl.DynamicProperty) [],
      userName = 'VSPHERE.LOCAL\\Administrator'
   },
   queueTime = 2019-08-23T10:27:50.948248Z,
   startTime = 2019-08-23T10:27:50.952971Z,
   completeTime = 2019-08-23T10:27:50.963454Z,
   eventChainId = 4824457,
   changeTag = <unset>,
   parentTaskKey = <unset>,
   rootTaskKey = <unset>,
   activationId = <unset>
}

In [14]:   
```

Reply:
gotcha. 
so, 
1. it returns you task and task has "state".
2. in the light of what you showed me are you sure this task is accomplished synchronously. otherwise, you have to wait for results then. same for create_folder.

Reply:
@izapolsk 
- I can improvise for destroy folder like this: ```wait_for(lambda: folder.Destroy().info.state == 'success', delay=2, timeout=10)```
- But I am not getting such info for create folder task:
```
In [29]: fd = w.create_folder("new_folder")                                                                                                                                  

In [30]: fd                                                                                                                                                                  
Out[30]: 'vim.Folder:group-v427'

In [31]: fd.info                                                                                                                                                             
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-31-9f47adab5bc8> in <module>
----> 1 fd.info

AttributeError: 'vim.Folder' object has no attribute 'info'

In [32]:  
```